From 97b0eef36c848293c2f680377abd6988cac56a84 Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Wed, 25 Jul 2012 08:23:05 +0200
Subject: [PATCH] libril: Provide support for legacy ril

With change Icc620fd191f8542f61bd38e365813753bfedefdf, RIL_VERSION_MIN was
increased from 2 to 6, but this causes problems for a number of devices
using ril blobs older than that.

  E/RILC    (  140): RIL_register: version 3 is to old, min version is 6

Consequently causing the ril system unable to connect to the rild socket.

Change-Id: I82c1cc4dab8fcaa4ccaa099132133b0f3b1d8029
---
 include/telephony/ril.h |    4 ++++
 libril/Android.mk       |    4 ++++
 2 files changed, 8 insertions(+)

diff --git a/include/telephony/ril.h b/include/telephony/ril.h
index 64952ca..990bc64 100644
--- a/include/telephony/ril.h
+++ b/include/telephony/ril.h
@@ -27,7 +27,11 @@
 #endif
 
 #define RIL_VERSION 7     /* Current version */
+#ifdef LEGACY_RIL
+#define RIL_VERSION_MIN 2 /* Minimum RIL_VERSION supported */
+#else
 #define RIL_VERSION_MIN 6 /* Minimum RIL_VERSION supported */
+#endif
 
 #define CDMA_ALPHA_INFO_BUFFER_LENGTH 64
 #define CDMA_NUMBER_INFO_BUFFER_LENGTH 81
diff --git a/libril/Android.mk b/libril/Android.mk
index fed7534..1f71cb5 100644
--- a/libril/Android.mk
+++ b/libril/Android.mk
@@ -29,6 +29,10 @@ ifeq ($(BOARD_USES_HC_RADIO),true)
 LOCAL_CFLAGS += -DHCRADIO
 endif
 
+ifeq ($(BOARD_USES_LEGACY_RIL),true)
+LOCAL_CFLAGS += -DLEGACY_RIL
+endif
+
 include $(BUILD_SHARED_LIBRARY)
 
 
-- 
1.7.10
